dart/lib/src/sentry_envelope.dart

```diff
@@ -0,0 +1,25 @@
+import 'sentry_envelope_header.dart';
+import 'sentry_envelope_item.dart';
+import 'protocol/sentry_event.dart';
+import 'protocol/sdk_version.dart';
+
+class SentryEnvelope {
+  SentryEnvelope(this.header, this.items);
+
+  final SentryEnvelopeHeader header;
+  final List<SentryEnvelopeItem> items;
+
+  static SentryEnvelope fromEvent(SentryEvent event, SdkVersion sdkVersion) {
+    return SentryEnvelope(SentryEnvelopeHeader(event.eventId, sdkVersion),
+        [SentryEnvelopeItem.fromEvent(event)]);
+  }
+
+  String serialize() {
```

Comment:
I think we need to make this async at some point.
As per [docs](https://develop.sentry.dev/sdk/envelopes/#size-limits) the Envelopes can get quite big and otherwise we could block the processing for too long.

Reply:
also for a JSON like this would work, but not when adding files (eg attachments), ideally we'd use a buffer/stream or something like that, that is only consumed when sending over the network, we'd need to guarantee that this works with the current `http` library though.

wondering if this would be required for that https://github.com/getsentry/sentry-dart/issues/392
looks like this lib has the capabilities https://api.dart.dev/stable/2.12.1/dart-io/HttpClientRequest-class.html

Reply:
but: Browser-based apps can't use this library :(

Reply:
`http` has a [StreamedRequest](https://pub.dev/documentation/http/latest/http/StreamedRequest-class.html) class. It should also work on web.

Reply:
Implemented the transport with `StreamedRequest`. `SentryEnvelope` now offers a `envelopeStream` method, which is then used to transfer chunked over the network. This is only done when gzip is disabled for requests.